### PR TITLE
Port new triple slash docs for .NET 7 Alpha [Dec '21 - Jan '22]

### PR DESCRIPTION
**Pull request by carlossanlop**
This PR is part of a monthly effort to port triple slash documentation to docs.

Area owners to review:

- Microsoft.Extensions.Hosting: @dotnet/area-extensions-hosting 
- System.Reflection.Metadata: @dotnet/area-system-reflection-metadata 
- System.Runtime.Intrinsics.X86: @dotnet/area-system-runtime-intrinsics 
- System.Threading: @mangod9 @kouvel

Past PRs:
- https://github.com/dotnet/dotnet-api-docs/pull/7495
- https://github.com/dotnet/dotnet-api-docs/pull/7424

cc @jeffhandley

### CODE REVIEW

**Review by eerhardt — APPROVED** [review]
Hosting LGTM

**Review by kouvel — COMMENTED** [review]
System.Threading LGTM

File: xml/Microsoft.Extensions.Hosting/BackgroundService.xml [hunk 1]
```diff
@@ -109,7 +109,15 @@
         <summary>This method is called when the <see cref="T:Microsoft.Extensions.Hosting.IHostedService" /> starts. The implementation should return a task that represents
             the lifetime of the long running operation(s) being performed.</summary>
         <returns>A <see cref="T:System.Threading.Tasks.Task" /> that represents the long running operations.</returns>
-        <remarks>To be added.</remarks>
+        <remarks>
+          <format type="text/markdown"><![CDATA[
+
+## Remarks
+
+See [Worker Services in .NET](https://docs.microsoft.com/dotnet/core/extensions/workers) for implementation guidelines.
```
Comment (gewarren) [inline xml/Microsoft.Extensions.Hosting/BackgroundService.xml:117]:
```suggestion
For implementation guidelines, see [Worker Services in .NET](/dotnet/core/extensions/workers).
```

File: xml/Microsoft.Extensions.Hosting/HostBuilder.xml [hunk 2]
```diff
@@ -76,7 +76,15 @@
       <Docs>
         <summary>Run the given actions to initialize the host. This can only be called once.</summary>
```
Comment (gewarren) [inline xml/Microsoft.Extensions.Hosting/HostBuilder.xml:77]:
```suggestion
        <summary>Runs the given actions to initialize the host. This method can only be called once.</summary>
```

File: xml/Microsoft.Extensions.Hosting/HostBuilder.xml [hunk 3]
```diff
@@ -76,7 +76,15 @@
...
         <returns>An initialized <see cref="T:Microsoft.Extensions.Hosting.IHost" />.</returns>
-        <remarks>To be added.</remarks>
+        <remarks>
+          <format type="text/markdown"><![CDATA[
+
+## Remarks
+
+Adds basic services to the host such as application lifetime, host environment, and logging.
```
Comment (gewarren) [inline xml/Microsoft.Extensions.Hosting/HostBuilder.xml:84]:
```suggestion
This method adds basic services to the host, such as application lifetime, host environment, and logging.
```

File: xml/System.Reflection.Metadata/AssemblyExtensions.xml [hunk 4]
```diff
@@ -64,12 +64,25 @@
...
-        <param name="length">To be added.</param>
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
-        <remarks>To be added.</remarks>
+        <param name="assembly">The assembly from which to retrieve the metadata.</param>
+        <param name="blob">When this method returns, contains the pointer to the metadata section blob.</param>
+        <param name="length">When this method returns, contains the length of the metadata section blob.</param>
+        <summary>Retrieves the metadata section of the assembly, for use with <see cref="T:System.Reflection.Metadata.MetadataReader" />.</summary>
+        <returns>
+          <see langword="true" /> if the metadata is retrieved successfully; <see langword="false" /> upon failure. The metadata might not be available for some assemblies, such as <see cref="T:System.Reflection.Emit.AssemblyBuilder" />, AOT images, etc.</returns>
```
Comment (gewarren) [inline xml/System.Reflection.Metadata/AssemblyExtensions.xml:72]:
```suggestion
          <see langword="true" /> if the metadata is retrieved successfully; <see langword="false" /> upon failure. The metadata might not be available for some assemblies, such as <see cref="T:System.Reflection.Emit.AssemblyBuilder" /> and AOT images.</returns>
```

**Review by gewarren — APPROVED** [review]

File: xml/Microsoft.Extensions.Hosting/BackgroundService.xml (hunk 1 above)
Comment (carlossanlop) [inline xml/Microsoft.Extensions.Hosting/BackgroundService.xml:117]: @gewarren I think I remember seeing that URLs from this domain can get the domain truncated. Is it the case with this one?

Comment (carlossanlop) [reply xml/Microsoft.Extensions.Hosting/BackgroundService.xml:117]: Nevermind, I see your review. GitHub fails to refresh the page or show comments when they are provided.

File: xml/System.Threading/CancellationToken.xml [hunk 5]
```diff
@@ -970,30 +970,31 @@ is captured along with the delegate and is used when executing it. The current <
...
+End If
+```
+
  ]]></format>
         </remarks>
         <exception cref="T:System.OperationCanceledException">The token has had cancellation requested.</exception>
         <related type="Article" href="/dotnet/standard/threading/cancellation-in-managed-threads">Cancellation</related>
         <related type="Article" href="/dotnet/standard/parallel-programming/task-cancellation">Task Cancellation</related>
         <related type="Article" href="/dotnet/standard/parallel-programming/how-to-cancel-a-task-and-its-children">How to: Cancel a Task and Its Children</related>
+        <exception cref="T:System.ObjectDisposedException">The associated <see cref="T:System.Threading.CancellationTokenSource"/> has been disposed.</exception>
```
Comment (buyaa-n) [inline xml/System.Threading/CancellationToken.xml:997]: It doesn't throw this exception and [i removed it recently](https://github.com/dotnet/dotnet-api-docs/pull/7555), apparently i need update the doc in the runtime repo

Comment (carlossanlop) [reply xml/System.Threading/CancellationToken.xml:997]: Thank you for letting me know. I'll revert the changes in this file.

**Review by buyaa-n — APPROVED** [review]
System.Reflection.Metadata LGTM

**Pull request merged**